File: src/main/java/com/vc/medium/L1004.scala
package com.vc.medium

object L1004 {
  def longestOnes(A: Array[Int], K: Int): Int = {
    var i = 0
    var j = 0
    var k = K
    val n = A.length
    while(i < n) {
      if(A(i) == 0) k -= 1
      if(k < 0) {
        if(A(j) == 0) k += 1
        j += 1
      }
      i += 1
    }
    i - j
  }
}

File: src/main/java/com/vc/medium/L1004.java
package com.vc.medium;

class L1004 {
    public int longestOnes(int[] A, int K) {
        int start = 0, end = 0, max = 0;
        while(end < A.length) {
            if(A[end++] == 0) K--;
            while(K < 0) {
                if(A[start++] == 0) K++;
            }
            max = Math.max(max, end - start);
        }
        return max;
    }
}